pkg/reference/reference.go

```diff
@@ -56,6 +56,22 @@ func ToPtrValue(v string) *string {
 	return &v
 }
 
+// FromPtrValues adapts a slice of string pointer fields for use as CurrentValues.
+func FromPtrValues(v []*string) (res []string) {
```

Comment:
Since length is known based on the passed slice we can pre-allocate the return slice before setting its values at each index. Would you mind updating to do so?

Reply:
Wondering if we should return `""` or skip the element when the value is `nil`. I guess putting `""` would be more expected by the users of this function, compared to giving 10 elements and retrieving, say, 3.

Reply:
Looks like we are now returning `""` which seems right to me because we want to error on that case in validation right? https://github.com/crossplane/crossplane-runtime/blob/8c9b0752894993fb456c2a438270c6d02dcf5b66/pkg/reference/reference.go#L159